.github/workflows/ci.yml

```
@@ -1,15 +1,43 @@
 name: CI
-on:
-  push: {}
-  pull_request: {}
+on: [push, pull_request]
 
 jobs:
   run:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - run: corepack enable
       - run: pnpm install
       - run: pnpm run build
       - run: pnpm run lint
       - run: pnpm test
+
+  test-types:
+    runs-on: ubuntu-latest
+    name: Test TypeScript Types with Node.js ${{ matrix.node }} and TypeScript ${{ matrix.ts }}
+    strategy:
+      matrix:
+        node: [20.x]
```

Let's keep the node version out of it until we want to test multiple node versions
```suggestion
    name: Test Types with TypeScript ${{ matrix.ts }}
    strategy:
      matrix:
```

---

Your wish is my command!